luarules/gadgets/unit_collision_damage_behavior.lua

```diff
@@ -0,0 +1,130 @@
+	unitsMaxImpulse[unitDefID] = unitDef.mass/maxImpulseThreshold
+	unitMasses[unitDefID] = unitDef.mass
+end
+
+for weaponDefID, wDef in ipairs(WeaponDefs) do
+	if wDef.damages and wDef.damages.impulseBoost and wDef.damages.impulseFactor then
+		weaponDefIDImpulses[weaponDefID] = {impulseboost = wDef.damages.impulseBoost, impulsefactor = wDef.damages.impulseFactor}
+	end
+end
+
+local function calculateImpulseData(unitDefID, damage, weaponDefID)
+	local impulse = (damage + weaponDefIDImpulses[weaponDefID].impulseboost) * weaponDefIDImpulses[weaponDefID].impulsefactor
+	local maxImpulse = unitsMaxImpulse[unitDefID]
+	local impulseMultiplier = mathmin(maxImpulse/impulse, 1)
+	return impulse, impulseMultiplier
+end
+
+local function isUnitBeingTransported(unitID)
+	for transportID, transportedID in pairs(transportingUnits) do
+		if transportingUnits[transportID][unitID] then
+			return true
+		end
+	end
+	return false
+end
+
+local function calculateMassToHealthRatioMultiplier(unitID, unitDefID)
+	local health, maxHealth = spGetUnitHealth(unitID)
+	if maxHealth then
+		local massToMaxHealthMultiplier = (maxHealth / unitMasses[unitDefID]) * fallDamageMultipliers[unitDefID]
+		return massToMaxHealthMultiplier
+	else
+		return fallDamageMultipliers[unitDefID]
+	end
+end
+
+function gadget:UnitPreDamaged(unitID, unitDefID, unitTeam, damage, paralyzer, weaponDefID, projectileID, attackerID, attackerDefID, attackerTeam)
+	if weaponDefID > 0 then --this section handles limiting maximum impulse
+		local impulseMultiplier = 1
+		local impulse = 0
+		if not overImpulseCooldowns[unitID] then
+			impulse, impulseMultiplier = calculateImpulseData(unitDefID, damage, weaponDefID)
+			overImpulseCooldowns[unitID] = {expireframe = currentModulusFrame + cooldownsFrameThreshold, highestimpulse = impulse, impulsemultiplier = impulseMultiplier}
+			return damage, impulseMultiplier
+		elseif overImpulseCooldowns[unitID] and overImpulseCooldowns[unitID].highestimpulse < impulse then
+			impulse, impulseMultiplier = calculateImpulseData(unitDefID, damage, weaponDefID)
+			impulseMultiplier = impulseMultiplier - overImpulseCooldowns[unitID].impulsemultiplier
+			overImpulseCooldowns[unitID] = {expireframe = currentModulusFrame + cooldownsFrameThreshold, highestimpulse = impulse, impulsemultiplier = impulseMultiplier}
+			return damage, impulseMultiplier
+		else
+			return damage, 0
+		end
+	else
+		if weaponDefID == groundCollisionDefID and not isUnitBeingTransported(unitID) then --handles ground collision events
+			damage = damage * calculateMassToHealthRatioMultiplier(unitID, unitDefID)
+			return damage, 0
+		elseif weaponDefID == objectCollisionDefID and not isUnitBeingTransported(unitID) then --handles object collision events
+			local _, velY, _, velLength = spGetUnitVelocity(unitID)
+			if velLength > objectCollisionVelocityThreshold and -velY > (velLength/3) then --prevents mostly horizontal object collisions from taking damage, allows damage if dropped from above
```

Comment:
1/3rd threshold does double duty. Prevents units from being launched sideways and bumping into each other from taking collision damage, and it prevents units from taking damage from a unit falling on them.

1/3rd is optimal ratio from my play testing. I don't see how making that a variable would make intent more clear.

I'll do the bool function tho

Reply:
I'm keeping the 1/3rd thing because declaring a variable will be more confusing IMO
but, I did make a bool function like you asked. Looks cleaner. Thanks.